src/Common/src/System/Runtime/ExceptionServices/ExceptionStackTrace.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+
+namespace System.Runtime.ExceptionServices
+{
+    internal static class ExceptionStackTrace
+    {
+        /// <summary>In debug builds, appends the current stack trace to the exception.</summary>
+        [Conditional("DEBUG")]
+        public static void AddCurrentStack(Exception exception)
+        {
+            Debug.Assert(exception != null, "Expected non-null Exception");
+
+            const string ExceptionRemoteStackTraceStringName = "_remoteStackTraceString";
+            FieldInfo fi = typeof(Exception).GetField(ExceptionRemoteStackTraceStringName, BindingFlags.NonPublic | BindingFlags.Instance);
+
+            if (fi != null)
+            {
+                string text =
+                    (string)fi.GetValue(exception) +
```

Comment:
I could.  I was just copying the existing logic that uses a single separator between entries (basically just adding an ending one rather than beginning).  Let's stick with this for now... easy enough to add a starting line subsequently if traces would be easier to read with it.